test/runner/lib/cloud/nios.py

```diff
@@ -0,0 +1,187 @@
+"""NIOS plugin for integration tests."""
+
+from __future__ import absolute_import, print_function
+
+import os
+
+from . import (
+    CloudProvider,
+    CloudEnvironment,
+)
+
+from ..util import (
+    find_executable,
+    display,
+)
+
+from ..docker_util import (
+    docker_run,
+    docker_rm,
+    docker_inspect,
+    docker_pull,
+    get_docker_container_id,
+)
+
+
+class NiosProvider(CloudProvider):
+    """Nios plugin.
+
+    Sets up NIOS mock server for tests.
+    """
+
+    DOCKER_SIMULATOR_NAME = 'nios-simulator'
+
+    DOCKER_IMAGE = 'quay.io/ansible/nios-test-container:latest'
```

Comment:
Tag a `1.0.0` release of the container and use that version here instead of `latest`.